app/src/main/java/com/netcetera/skopjepulse/countryCitySelector/CountryCityAdapter.kt

```diff
@@ -69,47 +69,44 @@ class CountryCityAdapter(var data: List<Any>?) : RecyclerView.Adapter<CountryCit
     return object : Filter() {
       override fun performFiltering(constraint: CharSequence?): FilterResults {
         val filterResults = FilterResults()
-        filterResults.values = CountryCityFilter.filterCountryCity(constraint,data)
+        filterResults.values = CountryCityFilter.filterCountryCity(constraint,context.getCountryCityViewModel().getSelectableCities())
         return filterResults
       }
 
       override fun publishResults(constraint: CharSequence?, results: FilterResults?) {
         dataShow = results?.values as MutableList<Any>
         notifyDataSetChanged()
       }
-
     }
   }
 
-
-
   abstract class BaseViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {
-    abstract fun bind(item: T)
+    abstract fun bind(item: T, countryName: String="")
   }
 
   class CityViewHolder(val view: View) : BaseViewHolder<CityItem>(view) {
-
-    private val cityCheckBox = view.checkBoxCity
-
-    override fun bind(item: CityItem) {
-      cityCheckBox.text = item.name
-      cityCheckBox.isChecked = item.isChecked
-
-      cityCheckBox.setOnClickListener {
-        item.isChecked = cityCheckBox.isChecked
+    private val cityItemRow = view.cityItemRow
+    private val cityName = view.txtCityName
+    private val countryName = view.txtCountryNameSmall
+
+    override fun bind(item: CityItem, countryName: String) {
+      cityName.text = item.name
+      this.countryName.text = countryName
+
+      cityItemRow.setOnClickListener{
+        item.isChecked = true
+        (view.context as CountryCitySelectorActivity).getCountryCityViewModel().saveCheckedCities()
+        (view.context as CountryCitySelectorActivity).finish()
```

Comment:
Suggest that we avoid such blind cast and non explicit communication `adapter -> activity`.

My suggestion is that to this adapter we pass callback function that is invoked by this action. Here is some example
```kt
class CountryCityAdapter(var data: List<Any>?, val onCitySelected : () -> void) {
//...
  cityItemRow.setOnClickListener {
        item.isChecked = true
        this.onCitySelected(item)
  }
}
```

Reply:
Fixed: [Fix onCitySelected](https://github.com/pulse-eco/pulse-eco-android/pull/5/commits/c2257586a60301cd5755976d58c94f0650092b5e)